geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/filesystem/FileSystem.java

```diff
@@ -60,7 +61,9 @@ public FileSystem(Map fileAndChunkRegion, FileSystemStats stats) {
 
   public Collection<String> listFileNames() {
     return (Collection<String>) fileAndChunkRegion.keySet().stream()
-        .filter(entry -> (entry instanceof String)).collect(Collectors.toList());
+        .filter(entry -> ((entry instanceof String) && !((String) entry)
```

Comment:
Perhaps create a ticket to explore whether we can use a non string value for the complete file.  I think there is a reason why this is needed but it might be nice explore and to be able to reduce the number of checks for clarity.